### PR TITLE
Add support to MacVtap interfaces

### LINKED ISSUE #1332: Add MacvTap interface support

### PR DESCRIPTION
**Pull request by ffmancera**
This patch needs a release from Nispor.

### CODE REVIEW

**Comment by ffmancera** [comment]
@cathay4t Ths is now ready for review. Thanks.

File: libnmstate/nm/plugin.py [hunk 1]
```diff
@@ -139,6 +140,9 @@ def get_interfaces(self):
             iface_info.update(nm_lldp.get_info(self.client, dev))
             iface_info.update(nm_team.get_info(dev))
             iface_info.update(get_infiniband_info(applied_config))
+            # This is a workaround needed until all the interface querying
+            # is done by Nispor.
```
Comment (cathay4t) [inline libnmstate/nm/plugin.py]: Nispor python binding already supported `NisporMacVtap`, while not use it?

**Review by cathay4t — CHANGES_REQUESTED** [review]
The `MacVtapIface` is 99% identical to `MacVlanIface`. How about use `class MacVtapIface(MacVlanIface)` and only redefined the difference?

The same duplication also exists for nispor and test case code.

File: libnmstate/nm/plugin.py (hunk 1 above)
Comment (ffmancera) [reply libnmstate/nm/plugin.py]: This is a workaround needed due to Nmstate gathering the interface type from NetworkManager, as we are deciding the interface type using the setting name. If the interface type is not adjusted, Nmstate will fail during verification as NM and Nispor interfaces will not be merged correctly.

I've added this explanation to the function.

**Comment by ffmancera** [comment]
> The `MacVtapIface` is 99% identical to `MacVlanIface`. How about use `class MacVtapIface(MacVlanIface)` and only redefined the difference?
> 
> The same duplication also exists for nispor and test case code.

Done, thank you!

File: libnmstate/ifaces/macvtap.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+from .macvlan import MacVlanIface
+
+
+class MacVtapIface(MacVlanIface):
+    @property
+    def parent(self):
+        return self._config.get(MacVtap.BASE_IFACE)
+
+    @property
+    def _config(self):
```
Comment (cathay4t) [inline libnmstate/ifaces/macvtap.py]: Redefine a internal property is not good.  Just change that to public interface as `config_subtree`.

File: libnmstate/nm/macvlan.py [hunk 3]
```diff
@@ -31,11 +34,10 @@
...
 
-def create_setting(iface_state, base_con_profile):
-    macvlan = iface_state.get(MacVlan.CONFIG_SUBTREE)
-    if not macvlan:
-        return None
-
+def create_setting(iface_state, base_con_profile, tap=False):
+    macvlan = iface_state.get(MacVlan.CONFIG_SUBTREE) or iface_state.get(
+        MacVtap.CONFIG_SUBTREE
+    )
```
Comment (cathay4t) [inline libnmstate/nm/macvlan.py:42]: `macvlan = iface_state.get(MacVtap.CONFIG_SUBTREE) if tap else iface_state.get(MacVlan.CONFIG_SUBTREE)`

File: libnmstate/nm/plugin.py (hunk 1 above)
Comment (cathay4t) [reply libnmstate/nm/plugin.py]: Then you need to reword the comment to say NM is require extra attention to get interface type right for mac_vtap and mac_vlan, it has nothing to do with nispor.

File: libnmstate/nm/profile.py [hunk 4]
```diff
@@ -579,7 +587,10 @@ def classify_profile_for_actions(self, groups):
                             ACTION_DEACTIVATE_BEFOREHAND
                         )
                         groups["profiles_to_deactivate_beforehand"].add(self)
-                elif self.iface.type == InterfaceType.MAC_VLAN:
+                elif (
+                    self.iface.type == InterfaceType.MAC_VLAN
+                    or self.iface.type == InterfaceType.MAC_VTAP
+                ):
```
Comment (cathay4t) [inline libnmstate/nm/profile.py]: A comment is required why and when it should be deactivated first.

And this seems not a good way to deactivate interface every time it has been included in desire state.

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: libnmstate/nm/plugin.py (hunk 1 above)
Comment (ffmancera) [reply libnmstate/nm/plugin.py]: I've moved the comment to the function and fixed the comment. Thanks!

File: libnmstate/ifaces/macvtap.py (hunk 2 above)
Comment (ffmancera) [reply libnmstate/ifaces/macvtap.py]: Done. Thank you!

File: libnmstate/nm/macvlan.py (hunk 3 above)
Comment (ffmancera) [reply libnmstate/nm/macvlan.py:42]: Done, thanks!

File: libnmstate/nm/profile.py (hunk 4 above)
Comment (ffmancera) [reply libnmstate/nm/profile.py]: I've added the comment and in addition I've added another condition to the statement. It is only being deactivated if the macvlan/macvtap interface is modified not only included in desired state.

**Review by cathay4t — APPROVED** [review]

**Pull request merged**